src/charm.py

```diff
@@ -9,76 +9,181 @@
         self.framework.observe(self.on.config_changed, self._on_config_changed)
 
         # MySQL
-        self.mysql = DatabaseRequires(self, relation_name="mysql", database_name=self.database)
+        self.mysql = DatabaseRequires(
+            self,
+            relation_name="mysql",
+            database_name=self.database,
+            extra_user_roles=self.extra_user_roles,
+        )
         self.framework.observe(self.mysql.on.database_created, self._on_database_created)
+        self.framework.observe(self.on["mysql"].relation_broken, self._on_database_relation_broken)
 
         # PostgreSQL
         self.postgresql = DatabaseRequires(
             self, relation_name="postgresql", database_name=self.database
         )
         self.framework.observe(self.postgresql.on.database_created, self._on_database_created)
+        self.framework.observe(
+            self.on["postgresql"].relation_broken, self._on_database_relation_broken
+        )
 
         # MongoDB
         self.mongodb = DatabaseRequires(self, relation_name="mongodb", database_name=self.database)
         self.framework.observe(self.mongodb.on.database_created, self._on_database_created)
+        self.framework.observe(
+            self.on["mongodb"].relation_broken, self._on_database_relation_broken
+        )
+
+        kafka_user_roles = KAFKA_EXTRA_USER_ROLES + self.extra_user_roles
+        self.kafka = KafkaRequires(
+            self, relation_name="kafka", topic=self.topic, extra_user_roles=kafka_user_roles
+        )
+        self.framework.observe(self.kafka.on.topic_created, self._on_topic_created)
+        self.framework.observe(self.on["kafka"].relation_broken, self._on_database_relation_broken)
 
         self._on_config_changed(None)
 
     def _on_config_changed(self, event):
-        self.database = self.model.config.get("database", "")
+
+        # database
+        new_database_name = self.model.config.get("database-name", "")
+        new_topic = self.model.config.get("topic-name", "")
+        new_extra_user_roles = self.model.config.get("extra-user-roles", "")
+
+        # database name not specified
         if self.database == "":
-            self.unit.status = BlockedStatus("The database name is not specified.")
-        elif event:
-            self.unit.status = ActiveStatus(f"database: {self.database}")
+            self.database == new_database_name
+        elif self.database != new_database_name:
+            self.unit.status = ActiveStatus(
+                f"New database name specified: {new_database_name}. Please remove current relation/s and add a new relation!"
+            )
+            return
+
+        if self.topic == "":
+            self.topic == new_topic
+        elif self.topic != new_topic:
+            self.unit.status = ActiveStatus(
+                f"New database name specified: {new_database_name}. Please remove current relation/s and add a new relation!"
```

Comment:
I'm not sure the UX is right here. I mean: I would probably go into either Maintenance or Block status, since the application is really not functioning. Probably Block is more correct, since we are expecting the user to do something.  

https://juju.is/docs/olm/status-values

Reply:
I substitute it with blocked status